server/src/main/java/org/opensearch/tasks/BackgroundTaskService.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.lifecycle.Lifecycle;
+import org.opensearch.common.lifecycle.LifecycleComponent;
+import org.opensearch.common.lifecycle.LifecycleListener;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.util.concurrent.AbstractAsyncTask;
+import org.opensearch.offline_tasks.TaskClient;
+import org.opensearch.offline_tasks.task.Task;
+import org.opensearch.offline_tasks.task.TaskType;
+import org.opensearch.offline_tasks.worker.TaskWorker;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Service to manage execution of tasks on Offline Node
+ */
+public class BackgroundTaskService implements LifecycleComponent {
```

Comment:
Need to add `@opensearch.internal`. 

Also, should we extend `AbstractLifecycleComponent` instead like the other services do, unless we plan to extend any other classes?

```suggestion
/**
 * Service to manage execution of tasks on Offline Node
 *
 * @opensearch.internal
 */
public class BackgroundTaskService extends AbstractLifecycleComponent {
```